src/libsync/logger.cpp

```diff
@@ -117,7 +118,33 @@ void Logger::doLog(QtMsgType type, const QMessageLogContext &ctx, const QString
             Utility::crash();
 #endif
         }
+        if (!_logDirectory.isEmpty()) {
+            if (_logFile.size() > maxLogSizeC) {
+                rotateLog();
+            }
+        }
+    }
+}
+
+void Logger::open(const QString &name)
+{
+    bool openSucceeded = false;
+    if (name == QLatin1Char('-')) {
+        attacheToConsole();
+        setLogFlush(true);
+        openSucceeded = _logFile.open(stdout, QIODevice::WriteOnly);
+    } else {
+        _logFile.setFileName(name);
+        openSucceeded = _logFile.open(QIODevice::WriteOnly);
+    }
```

Comment:
Might be easier to just move that into a lambda (like it is done with a lot of variable initializations).